ci/build_scripts/build.sh

```diff
@@ -0,0 +1,174 @@
+
+        * x86_64-unknown-linux-gnu
+        * aarch64-unknown-linux-gnu
+        * aarch64-unknown-linux-musl
+        * arm-unknown-linux-gnueabihf
+        * armv7-unknown-linux-gnueabihf
+        * armv7-unknown-linux-musleabihf
+
+Flags:
+    --use-cross     Force to use cross to build the packages
+
+Examples:
+    $0
+    # Build for the current CPU architecture
+
+    $0 aarch64-unknown-linux-gnu
+    # Build for arm64 linux (gnu lib)
+
+    $0 x86_64-unknown-linux-gnu
+    # Build for x86_64 linux (gnu lib)
+
+    $0 armv7-unknown-linux-gnueabihf
+    # Build for armv7 (armhf) linux (gnu lib)
+
+    $0 --use-cross
+    # Force to use cross when building for the current architecture
+EOF
+}
+
+ARCH=
+TARGET=()
+
+REST_ARGS=()
+while [ $# -gt 0 ]
+do
+    case "$1" in
+        -c|--use-cross)
+            USE_CROSS=1
+            ;;
+
+        -h|--help)
+            help
+            exit 0
+            ;;
+        
+        *)
+            REST_ARGS+=("$1")
+            ;;
+    esac
+    shift
+done
+set -- "${REST_ARGS[@]}"
+
+if [ $# -eq 1 ]; then
+    ARCH="$1"
+fi
+
+# Install required cargo crates
+cargo install cargo-deb --version 1.38.1
+
+# Detect current host, and use cross to build if the target does not match the current host arch (triple)
+HOST_ARCH=$(rustc -vV | sed -n 's|host: ||p')
+USE_CROSS=${USE_CROSS:-}
+
+if [ -z "$USE_CROSS" ]; then
+    if [[ -n "$ARCH" && "$HOST_ARCH" != "$ARCH" ]]; then
+        USE_CROSS=1
+    else
+        USE_CROSS=0
+    fi
+fi
+
+BUILD_CMD=cargo
+if [ "$USE_CROSS" == "1" ]; then
+    cargo install cross
+    echo "Using cross to compile binaries"
+    BUILD_CMD=cross
+fi
+
+if [ -n "$ARCH" ]; then
+    TARGET+=("--target=$ARCH")
+fi
+
+# Install libc/libgcc dependencies for cross compiling and building the deb package
+case "$ARCH" in
+    # e.g. armv5
+    arm-unknown-linux-gnueabi)
+        sudo apt-get -y update
+        sudo apt-get -y install gcc-arm-linux-gnueabi
+        ;;
+
+    # e.g. armv6 (Raspberry Pi Zero)
+    arm-unknown-linux-gnueabihf)
+        sudo apt-get -y update
+        sudo apt-get -y install -qq gcc-arm-linux-gnueabihf libc6-armhf-cross libc6-dev-armhf-cross
+        ;;
+
+    armv7-unknown-linux-gnueabihf)
+        sudo apt-get -y update
+        sudo apt-get -y install gcc-arm-linux-gnueabihf
+        ;;
+
+    aarch64-unknown-linux-gnu)
+        sudo apt-get -y update
+        sudo apt-get -y install gcc-aarch64-linux-gnu
+        ;;
+
+    x86_64-unknown-linux-*)
+        sudo apt-get -y update
+        sudo apt-get -y install gcc-x86-64-linux-gnu
+        ;;
+esac
+
+
+# Load the release package list as $RELEASE_PACKAGES and $TEST_PACKAGES
+# shellcheck disable=SC1091
+source ./ci/package_list.sh
+
+# build release for target
+"$BUILD_CMD" build --release "${TARGET[@]}"
+
+# Work out which strip command
+# Figure out what strip tool to use if any
+# Figure out what strip tool to use if any
+STRIP=
+if [ -n "$ARCH" ]; then
+    STRIP="strip"
```

Comment:
If user provides an architecture which is not listed below, `$STRIP` has `strip` as value. Then, line 160,
```
"$STRIP" target/"$ARCH"/release/"$PACKAGE"
``` 
it will fail. Is it intended?

Reply:
There is a `strip` command which is the for the current cpu architecture, so it should work fine.